---
title: "Identification of differentially methylated CpGs sites"
author: "Sabin Hart"
date: "`r Sys.Date()`"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries
```{r}
# BiocManager::install("ChAMP")
# install.packages("Rideogram")
# install.packages("UpSetR")

library(ChAMP)
library(RIdeogram)
library(UpSetR)
library(ChAMP)
library(minfi)  
library(Biobase)  
library(GEOquery)
library(EnhancedVolcano)
```
# Load Data
```{r}
geo_data <- getGEO("GSE68319", GSEMatrix = TRUE)
methyl_data <- geo_data[[1]]
```


# 1.
DMCs were obtained by comparing the methylation sites of VV and normal samples
in the GSE68319 dataset using the "ChAMP" (version 2.20.1) R package with a 
screening condition of p < 0.05 

```{r}
beta_values <- exprs(methyl_data)
pheno_data <- pData(methyl_data)
pheno_data$condition <- ifelse(grepl("varicose", pheno_data$description), 1, 0)

myLoad <- list(beta = beta_values, pd = pheno_data)

DMPs <- champ.DMP(beta = myLoad$beta, pheno = pheno_data$condition, adjust.method="none")

DMCs <- DMPs[[1]]

write.csv(DMCs, "DMCs.csv")
```

# Figure 1A - Volcano Plot
```{r}
png("volcano.png", width = 800, height = 1000)  
EnhancedVolcano(DMCs,
    lab = rownames(DMCs),
    x = 'logFC',
    y = 'P.Value',
    FCcutoff = 0,
    title = "VV vs. Normal",
    legendPosition="right",
    xlim = c(-0.2, 0.2),
    ylim = c(1.3, 8.5))
dev.off()
```



# 2.
The "Rideogram" R package (version 0.2.2) was then used to visualize the 
chromosome annotation information of DMCs 
```{r}
#data("human_karyotype",package="RIdeogram")
#ideogram(karyotype = human_karyotype)
#convertSVG("chromosome.svg", device="png")

rideogram_data <- data.frame(
  Chr = DMCs$CHR,  
  Start = DMCs$MAPINFO,                
  End = DMCs$MAPINFO + 1,              
  Value = 1000*DMCs$P.Value                    
)


# Pass the karyotype and feature data to RIdeogram
ideogram(karyotype = human_karyotype, overlaid =  rideogram_data)
convertSVG("chromosome.svg", device = "png")
```


# 3. 
Finally, we compared the distribution of CpG sites in different regions within 
genes to obtain differentially methylated regions (DMRs) using the "UpSetR"
(version 1.4.0) R package
```{r}
png("pie_chart.png", width = 800, height = 800)  
pie(table(DMCs$feature), col = c("skyblue", "firebrick", "orange", "deeppink", "purple", "lightgreen", "khaki"))
dev.off()
```

