---
title: "Functional enrichment analysis of methylation-regulated genes"
author: "Sabin Hart"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries
```{r, echo=FALSE}
library(GEOquery)
library(tidyverse)
#BiocManager::install("edgeR")
library(edgeR)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)
library(illuminaHumanv4.db)
```
# Load Data
```{r}
geo_data <- getGEO("GSE68309", GSEMatrix = TRUE)
expres_dat <- exprs(geo_data[[1]])
```

In the GSE68309 dataset, differentially expressed genes (DEGs) between VV and 
normal samples were compared using the “edgeR” (version 3.34.1) R package with 
the screening condition of |log2FC| > 0.5, p < 0.05 

# segregate samples
```{r}
pheno <- pData(geo_data[[1]])
group <- factor(if_else(grepl("Varicose", pheno$title), "VV", "Control"))

dge <- DGEList(counts = expres_dat, group = group)
keep <- filterByExpr(dge)
dge <- dge[keep, , keep.lib.sizes=FALSE]

# normalize
dge <- calcNormFactors(dge)
```

# differential expression analysis
```{r}
design <- model.matrix(~ group)
dge <- estimateDisp(dge, design)

# fit and transform model
fit <- glmQLFit(dge, design)
result <- glmQLFTest(fit, coef=2)  # Adjust the coefficient if needed

# screening condition to get DEGs
degs <- topTags(result, n=Inf)$table
degs_filtered <- subset(degs, abs(logFC) > 0.5 & PValue < 0.05) 

degs_filtered <- degs_filtered %>% 
  mutate("IlluminaID" = rownames(degs_filtered))

# replace Illumina IDs with gene names
gene_symbols <- mget(x = rownames(degs_filtered),
                     envir = illuminaHumanv4SYMBOL, ifnotfound = NA)
gene_symbols <- unlist(gene_symbols)
gene_symbols[is.na(gene_symbols)] <- rownames(degs_filtered)[is.na(gene_symbols)]
gene_symbols <- make.unique(gene_symbols)
rownames(degs_filtered) <- gene_symbols



# View top differentially expressed genes
head(degs_filtered)
```

# visualize
```{r}
# volcano plot
png("volcano2.png", width = 1200, height = 1000)  
EnhancedVolcano(degs,
    lab = rownames(degs),
    x = 'logFC',
    y = 'PValue',
    FCcutoff = 0.5,
    pCutoff = 0.05,
    title = "VV vs. Normal",
    legendPosition="right",
    xlim = c(-2, 2),
    ylim = c(0, 5))
dev.off()

# heatmap
degs_matrix <- expres_dat[rownames(expres_dat) %in% degs_filtered$IlluminaID, ]
gene_symbols <- mget(x = rownames(degs_matrix),
                     envir = illuminaHumanv4SYMBOL, ifnotfound = NA)
gene_symbols <- unlist(gene_symbols)
gene_symbols[is.na(gene_symbols)] <- rownames(degs_matrix)[is.na(gene_symbols)]
gene_symbols <- make.unique(gene_symbols)
rownames(degs_matrix) <- gene_symbols
scaled <- t(scale(t(degs_matrix)))  # Standardize across genes

my_palette <- colorRampPalette(c("blue", "white", "red"))(100)
sample_annotation <- data.frame(Group = group)
rownames(sample_annotation) <- colnames(expres_dat)  # Match column names

png("heatmap.png", width = 1000, height = 800)  
pheatmap(scaled,
         color = my_palette,  
         cluster_rows = TRUE,   # Cluster genes
         cluster_cols = FALSE,  # Do NOT cluster samples
         annotation_col = sample_annotation,  # Add sample grouping
         show_rownames = TRUE,  
         show_colnames = FALSE,  
         scale = "row",  
         fontsize_row = 10)
dev.off()

```

# Libraries
```{r, echo=FALSE}
library(clusterProfiler)
library(org.Hs.eg.db)  # For human gene annotation
```

# Part 2
```{r}
DEGs <- degs_filtered
DEGs$gene <- rownames(DEGs)
DMCs <- read_csv("../Method 1/DMCs.csv")

# Categorize genes based on methylation changes
hypermethylated_genes <- DMCs %>%
  filter(logFC > 0) %>%
  pull(gene) %>% unique()

demethylated_genes <- DMCs %>%
  filter(logFC < 0) %>%
  pull(gene) %>% unique()


# get VV-MRGs
MRGs_hyper_low <- intersect(hypermethylated_genes, DEGs %>% filter(logFC < 0) %>% pull(gene))
MRGs_hypo_high <- intersect(demethylated_genes, DEGs %>% filter(logFC > 0) %>% pull(gene))

# Combine MRG lists
MRGs <- unique(c(MRGs_hyper_low, MRGs_hypo_high))

MRGs

MRGs_entrez <- bitr(MRGs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Hs.eg.db")$ENTREZID
```

# visualize
```{r}
# GO
GO_results <- enrichGO(
  gene = MRGs_entrez,
  OrgDb = org.Hs.eg.db,
  keyType = "ENTREZID",
  ont = "BP",  # Can also try "BP", "CC", or "MF"
  pAdjustMethod = "BH",
  pvalueCutoff = 0.05
)

# Plot results
barplot(GO_results, showCategory = 10, title = "GO Enrichment Analysis")



# KEGG
KEGG_results <- enrichKEGG(
  gene = MRGs_entrez,
  organism = "hsa",
  pAdjustMethod = "none",
  pvalueCutoff = 0.05
)


# Plot results
barplot(KEGG_results, showCategory = 10, title = "KEGG Enrichment Analysis")


```


