---
title: "Functional enrichment analysis of methylation-regulated genes"
author: "Sabin Hart"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries
```{r}
library(GEOquery)
library(tidyverse)
#BiocManager::install("edgeR")
library(edgeR)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)

```
# Load Data
```{r}
geo_data <- getGEO("GSE68309", GSEMatrix = TRUE)
expres_dat <- exprs(geo_data[[1]])
```

In the GSE68309 dataset, differentially expressed genes (DEGs) between VV and 
normal samples were compared using the “edgeR” (version 3.34.1) R package with 
the screening condition of |log2FC| > 0.5, p < 0.05 

# segregate samples
```{r}
pheno <- pData(geo_data[[1]])
group <- factor(if_else(grepl("Varicose", pheno$title), "VV", "Control"))

dge <- DGEList(counts = expres_dat, group = group)
keep <- filterByExpr(dge)
dge <- dge[keep, , keep.lib.sizes=FALSE]

# normalize
dge <- calcNormFactors(dge)
```

# differential expression analysis
```{r}
design <- model.matrix(~ group)
dge <- estimateDisp(dge, design)

# fit and transform model
fit <- glmQLFit(dge, design)
result <- glmQLFTest(fit, coef=2)  # Adjust the coefficient if needed

# screening condition to get DEGs
degs <- topTags(result, n=Inf)$table
degs_filtered <- subset(degs, abs(logFC) > 0.5 & PValue < 0.05)

# View top differentially expressed genes
head(degs_filtered)
```

```{r}

EnhancedVolcano(degs,
    lab = rownames(degs),
    x = 'logFC',
    y = 'PValue',
    FCcutoff = 0.5,
    pCutoff = 0.05,
    title = "VV vs. Normal",
    legendPosition="right")

    xlim = c(-0.2, 0.2),
    ylim = c(1.3, 8.5))

ggplot(degs, aes(x=logFC, y=-log10(PValue))) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=c(-0.5, 0.5), linetype="dashed", color="red") +
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color="blue") +
  theme_minimal() +
  labs(title="Volcano Plot of DEGs", x="log2 Fold Change", y="-log10 P-Value")

```


